src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java

```diff
@@ -0,0 +1,155 @@
+
+    /**
+     * String indicators of parameter in .obj file.
+     */
+
+    private final static String OBJ_VERTEX_NORMAL = "vn";
+    private final static String OBJ_VERTEX = "v";
+    private final static String OBJ_FACE = "f";
+
+    private ArrayList<Vector3> verticesNormals = new ArrayList<>();
+    private ArrayList<Vector3> verticesGeometry = new ArrayList<>();
+    private PolygonObject polygonObject = null;
+
+    File objFile = null;
+
+
+    public ParseObjFile(String filename) throws FileNotFoundException, IOException {
+
+        polygonObject = new PolygonObject();
+        parseObjFile(filename);
+
+    }
+
+    /**
+     * This part pars object file.
+     *
+     * @param objFilename
+     * @throws FileNotFoundException
+     * @throws IOException
+     */
+    private void parseObjFile(String objFilename) throws FileNotFoundException, IOException {
+        int lineCount = 0;
+        InputStreamReader fileReader;
+        BufferedReader bufferedReader;
+
+        objFile = new File(objFilename);
+        fileReader = new InputStreamReader(new FileInputStream(objFilename), "UTF-8");
+        bufferedReader = new BufferedReader(fileReader);
+
+        String line;
+
+        // read .obj file
+        while (true) {
+            line = bufferedReader.readLine();
+            if (null == line) {
+                break;
+            }
+
+            line = line.trim();
+
+            if (line.length() == 0) {
+                continue;
+            }
+
+            // cases to process
+            if (line.startsWith("#")) {
+                continue;
+            } else if (line.startsWith(OBJ_VERTEX_NORMAL)) {
+                processVertexNormal(line);
+                // add vertex with some coordinates
+            } else if (line.startsWith(OBJ_VERTEX)) {
+                processVertex(line);
+            } else if (line.startsWith(OBJ_FACE)) {
+                processFace(line);
+            } else {
+                log.log(WARNING, "line " + lineCount + " unknown line |" + line + "|");
+            }
+            lineCount++;
+        }
+        bufferedReader.close();
+
+        log.log(INFO, "Loaded " + lineCount + " lines");
+    }
+
+
+    public PolygonObject getPolygonObject() {
+        return polygonObject;
+    }
+
+    /**
+     * Create vertex( just coordinates of each one) from data of file.
+     *
+     * @param line
+     */
+
+    private void processVertex(String line) {
+        String[] values = line.split(" ");
+        float[] result = new float[3];
+        for (int i = 2; i < values.length; i++) {
+            result[i - 2] = Float.parseFloat(values[i]);
+        }
+        verticesGeometry.add(new Vector3(result[0], result[1], result[2]));
+    }
+
+
+    /**
+     * Create normal-data of vertex
+     *
+     * @param line
+     */
+    private void processVertexNormal(String line) {
+        String[] values = line.split(" ");
```

Comment:
This code block repeats. Maybe create `parseLineToVector3(String line, int offset)`?